vm/object.c

```diff
@@ -64,27 +65,47 @@ int vm_objectGet(vm_object_t **o, oid_t oid)
 	*o = lib_treeof(vm_object_t, linkage, lib_rbFind(&object_common.tree, &t.linkage));
 
 	if (*o == NULL) {
-		sz = (size_t)proc_size(oid);
-		n = round_page(sz) / SIZE_PAGE;
+		/* Take off the lock to avoid a deadlock in vm_kmalloc */
+		proc_lockClear(&object_common.lock);
 
-		if ((*o = (vm_object_t *)vm_kmalloc(sizeof(vm_object_t) + n * sizeof(page_t *))) == NULL) {
-			proc_lockClear(&object_common.lock);
-			return -ENOMEM;
+		sz = proc_size(oid);
+		if (sz < 0) {
+			err = (int)sz;
```

Comment:
Shouldn't we exit the function at this point?

Reply:
We'll just try to find the object in the tree again and exit in line 86 if we fail again. It's a feature @ziemleszcz requested and it doesn't really hurt anything